cfme/utils/ssh.py

```diff
@@ -323,58 +324,36 @@ def _run_command(self, command, timeout=RUNCMD_TIMEOUT, reraise=False, ensure_ho
             if timeout:
                 session.settimeout(float(timeout))
 
-            session.exec_command(command)
-            stdout = session.makefile()
-            stderr = session.makefile_stderr()
-
             def write_output(line, file):
                 output.append(line)
                 if self._streaming:
                     file.write(line)
 
-            while True:
-                if session.exit_status_ready():
+            session.exec_command(command)
+            bufsize = 1024
+            stdin = session.makefile('wb', bufsize)
+            stdout = session.makefile('r', bufsize)
+            stderr = session.makefile_stderr('r', bufsize)
+            channel = stdout.channel
+
+            stdin.close()
+            channel.shutdown_write()
+            while not channel.closed:
+                readq, _, _ = select.select([channel], [], [], timeout)
+                for c in readq:
+                    if c.recv_ready():
+                        write_output(channel.recv(len(c.in_buffer)), self.f_stdout)
+                    if c.recv_stderr_ready():
+                        write_output(channel.recv(len(c.in_buffer)), self.f_stderr)
```

Comment:
**required** the correct apis here are `in_stderr_buffer` and `recv_stderr`

Reply:
**OPTIONAL** the need for the channel files got removed, i believe its absolutely fine just to go for the channel